### PR TITLE
Otlp http exporter: allow endpoint override

### PR DESCRIPTION
**Pull request by rypdal**
Fixes #.

## Changes

Implemented the following behaviour:

[default path](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp) is only appended when the Options.Endpoint property wasn't set by the user, the protocol is HttpProtobuf and the OTEL_EXPORTER_OTLP_ENDPOINT environment variable is present. If the user provides a custom value for Options.Endpoint that value is taken as is.

For significant contributions please make sure you have completed the following items:

* [ ] `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by rypdal** [comment]
> @rypdal I have a few questions regarding spec compliance here . The [spec ](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#endpoint-urls-for-otlphttp) mentions two scenarios for constructing URLs:
> 
> > 1. For the per-signal variables (OTEL_EXPORTER_OTLP__ENDPOINT), the URL MUST be used as-is without any modification. The only exception is that if an URL contains no path part, the root path / MUST be used (see Example 2).
> 
> It looks like currently we are not reading from `OTEL_EXPORTER_OTLP_TRACES_ENDPOINT` or `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT`. Is this planned for a future PR? Also, these variables should allow one to override the URL for sending a specific signal as shown in [Example 3](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/protocol/exporter.md#example-3) of the spec. With the changes in this PR, the `OtlpExporterOptions.TracesEndpoint` and `OtlpExporterOptions.MetricsEndpoint` cannot have any value different from `OtlpExporterOptions.Endpoint`. Is this not expected to be covered by this PR?
> 
> > 1. If signals are sent that have no per-signal configuration from the previous point, OTEL_EXPORTER_OTLP_ENDPOINT is used as a base URL and the signals are sent to these paths relative to that:
> > 
> > 
> > * Traces: v1/traces
> > * Metrics: v1/metrics.
> > 
> > Non-normatively, this could be implemented by ensuring that the base URL ends with a slash and then appending the relative URLs as strings.
> 
> When someone sets the `OtlpExporterOptions.Endpoint` property, do we not want to append "v1/traces" or "v1/metrics" to `OtlpExporterOptions.TracesEndpoint` and `OtlpExporterOptions.MetricsEndpoint` if the `Protocol` is `HttpProtobuf`?

@utpilla , please, see my answers/last changes below:

1. I decided to remove traces and metrics endpoint properties from exporter options as it should be considered along with per-signal env. vars. Moreover, other per-signal env. vars. like OTEL_EXPORTER_OTLP_{Signal}_HEADERS, OTEL_EXPORTER_OTLP_{Signal}_PROTOCOL, etc should be considered in dedicated PR as well. The scope of this PR is only allow endpoint override. AFAIK PR for support of signal specific env. vars. not exists yet. 
2. Current implementation is changed in the way that traces and metrics relative paths append logic is moved to corresponding AddOtlpExporter methods where call scope (traces or metrics) is known.
3. Specification is clear about appending of "v1/traces" or "v1/metrics" only for case when endpoint is read from OTEL_EXPORTER_OTLP_ENDPOINT. If user overwrites Endpoint then it stays as is without any changes. I think the main subject of [this](https://github.com/open-telemetry/opentelemetry-dotnet/issues/2483) issue is to allow user to overwrite with it's own export endpoint.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs [hunk 1]
```diff
@@ -131,5 +131,70 @@ public void AppendPathIfNotPresent_TracesPath_AppendsCorrectly(string inputUri,
...
+
+            options.AppendExportPath(options.Endpoint, "test/path");
+
+            Assert.Equal("http://localhost:4317/", options.Endpoint.AbsoluteUri);
+        }
+
+        [Fact]
+        public void AppendExportPath_EndpointNotSet_EnvironmentVariableDefined_Adjusted()
+        {
+            ClearEndpointEnvVar();
```
Comment (mic-max) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs:150]:
```suggestion
```

The clearing of the environment is unnecessary as it is immediately set to a value

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs [hunk 2]
```diff
@@ -131,5 +131,70 @@ public void AppendPathIfNotPresent_TracesPath_AppendsCorrectly(string inputUri,
...
+
+            Assert.Equal("http://test:8888/test/path", options.Endpoint.AbsoluteUri);
+
+            ClearEndpointEnvVar();
+        }
+
+        [Fact]
+        public void AppendExportPath_EndpointSetEqualToEnvironmentVariable_EnvironmentVariableDefined_NotAdjusted()
+        {
+            ClearEndpointEnvVar();
```
Comment (mic-max) [inline test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs:165]:
```suggestion
```

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs (hunk 1 above)
Comment (rypdal) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs]: Removed.

File: test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs (hunk 2 above)
Comment (rypdal) [reply test/OpenTelemetry.Exporter.OpenTelemetryProtocol.Tests/OtlpExporterOptionsExtensionsTests.cs]: Removed.

**Review by alanwest — APPROVED** [review]
LGTM, thanks @rypdal!

**Comment by rypdal** [comment]
@cijothomas , @pellared , @alanwest ,  @utpilla , @joaopgrassi. @mic-max , @reyang thank you all for the review and discussions.

**Pull request merged**